### PR TITLE
GFX: sort versions, flags to return the highest version we support in…

### PR DESCRIPTION
**Pull request by jsorg71**
… caps advertise
I have run into some clients(mainly older mstsc) that did not work with GFX because missing capabilities.

### CODE REVIEW

File: xrdp/xrdp_mm.c [hunk 1]
```diff
@@ -1218,15 +1246,27 @@ xrdp_mm_egfx_caps_advertise(void *user, int caps_count,
...
+                {
+                    best_h264_index = index;
+                }
                 best_pro_index = index;
                 break;
             case XR_RDPGFX_CAPVERSION_101:
                 best_pro_index = index;
                 break;
             case XR_RDPGFX_CAPVERSION_102:
+                if (!(flags & XR_RDPGFX_CAPS_FLAG_AVC_DISABLED))
```
Comment (Nexarian) [inline xrdp/xrdp_mm.c]: The code for 102, 103, 105, 106, and 107 all look identical... Perhaps we just consolidate those into one "fall through" switch block?

**Comment by Nexarian** [comment]
Great fix! One comment that we could probably consolidate the switch statements to be less DIY-ey

File: xrdp/xrdp_mm.c (hunk 1 above)
Comment (Nexarian) [reply xrdp/xrdp_mm.c]: 8 & 101 are the same as well.

File: xrdp/xrdp_mm.c [hunk 2]
```diff
@@ -1218,29 +1247,26 @@ xrdp_mm_egfx_caps_advertise(void *user, int caps_count,
...
+                {
+                    best_h264_index = index;
+                }
                 best_pro_index = index;
                 break;
-            case XR_RDPGFX_CAPVERSION_106:
-                best_pro_index = index;
+            default:
+                /* just skip unknwown */
```
Comment (Nexarian) [inline xrdp/xrdp_mm.c:1269]: Add a log statement that we found an unknown cap, probably this should be LOG(LOG_INFO...)

**Comment by jsorg71** [comment]
Yea, my mac client lists caps version 0x000b0101, I don't know what that is.

**Comment by Nexarian** [comment]
Ask ChatGPT!

**Comment by jsorg71** [comment]
hehe

Jay
what is RDP GFX version 0x000b0101

ChatGPT
The Remote Desktop Protocol (RDP) GFX version 0x000b0101 refers to a specific version of the graphics capabilities in the RDP protocol. In this context:

"0x000b" indicates the major version.
"0101" indicates the minor version.

**Pull request merged**